### PR TITLE
Handful of graph-v2 fixes

### PR DESCRIPTION
**Pull request by edvald**
See individual commits. The first one was the sneakiest one, made it seem more like a solver issue, and masked actual underlying errors.

The 2nd commit is worth noting since that fix probably will break some other tests in a superficial manner. From the commit message:

| Note: We used to include container schema fields in the unit test plugin, which was confusing to begin with. It's now only based on the exec plugin.

Take a look @Orzelius, lmk if this all makes sense.

### CODE REVIEW

File: core/test/unit/src/logger/logger.ts [hunk 1]
```diff
@@ -230,4 +233,171 @@ describe("Logger", () => {
...
+      const obj = {
+        a: logger.info("foo"),
+      }
+      const res = sanitizeValue(obj)
+      expect(res).to.eql({
+        a: "<LogEntry>",
+      })
+    })
+
+    it("calls sanitize method if present", async () => {
```
Comment (vvagaytsev) [inline core/test/unit/src/logger/logger.ts:365]: It looks like this test case duplicates the [previous one](https://github.com/garden-io/garden/pull/3457/files#diff-a15e291e1abfd8502e234bb0d238e333bd9166a669b2937c1fd55ee315a70b60R355).

**Comment by vvagaytsev** [comment]
LGTM 👍 . I left a few minor comments, please check.

File: core/test/helpers.ts [hunk 2]
```diff
@@ -94,13 +97,14 @@ export const projectRootA = getDataDir("test-project-a")
 export const projectRootBuildDependants = getDataDir("test-build-dependants")
 export const projectTestFailsRoot = getDataDir("test-project-fails")
 
-const testModuleTestSchema = () => containerModuleTestSchema().keys({ command: joi.sparseArray().items(joi.string()) })
+const testModuleTestSchema = () => execTestSchema().keys({ command: joi.sparseArray().items(joi.string()) })
 
-const testModuleTaskSchema = () => containerTaskSchema().keys({ command: joi.sparseArray().items(joi.string()) })
+const testModuleTaskSchema = () => execTaskSpecSchema().keys({ command: joi.sparseArray().items(joi.string()) })
```
[the remark this answers is not in the document]
Comment (edvald) [reply core/test/helpers.ts:102]: Good catch, that is indeed redundant. I'll clean that up.

**Comment by edvald** [comment]
Resolved the comments @vvagaytsev

**Review by vvagaytsev — APPROVED** [review]

**Pull request merged**